specparam/objs/data.py

```diff
@@ -0,0 +1,475 @@
+            spectra_dim == 3 and freqs.shape[-1] != powers.shape[1]:
+            raise InconsistentDataError("The input frequencies and power spectra "
+                                        "are not consistent size.")
+
+        # Check if power values are complex
+        if np.iscomplexobj(powers):
+            raise DataError("Input power spectra are complex values. "
+                            "Model fitting does not currently support complex inputs.")
+
+        # Force data to be dtype of float64
+        #   If they end up as float32, or less, scipy curve_fit fails (sometimes implicitly)
+        if freqs.dtype != 'float64':
+            freqs = freqs.astype('float64')
+        if powers.dtype != 'float64':
+            powers = powers.astype('float64')
+
+        # Check frequency range, trim the power values range if requested
+        if freq_range:
+            freqs, powers = trim_spectrum(freqs, powers, freq_range)
+
+        # Check if freqs start at 0 and move up one value if so
+        #   Aperiodic fit gets an inf if freq of 0 is included, which leads to an error
+        if freqs[0] == 0.0:
+            freqs, powers = trim_spectrum(freqs, powers, [freqs[1], freqs.max()])
+            if self.verbose:
+                print("\nFITTING WARNING: Skipping frequency == 0, "
+                      "as this causes a problem with fitting.")
+
+        # Calculate frequency resolution, and actual frequency range of the data
+        freq_range = [freqs.min(), freqs.max()]
+        freq_res = freqs[1] - freqs[0]
+
+        # Log power values
+        powers = np.log10(powers)
+
+        ## Data checks - run checks on inputs based on check modes
+
+        if self._check_freqs:
+            # Check if the frequency data is unevenly spaced, and raise an error if so
+            freq_diffs = np.diff(freqs)
+            if not np.all(np.isclose(freq_diffs, freq_res)):
+                raise DataError("The input frequency values are not evenly spaced. "
+                                "The model expects equidistant frequency values in linear space.")
+        if self._check_data:
+            # Check if there are any infs / nans, and raise an error if so
+            if np.any(np.isinf(powers)) or np.any(np.isnan(powers)):
+                error_msg = ("The input power spectra data, after logging, contains NaNs or Infs. "
+                             "This will cause the fitting to fail. "
+                             "One reason this can happen is if inputs are already logged. "
+                             "Input data should be in linear spacing, not log.")
+                raise DataError(error_msg)
+
+        return freqs, powers, freq_range, freq_res
+
+
+class BaseData2D(BaseData):
+    """Base object for managing data for spectral parameterization - for 2D data."""
+
+    def __init__(self):
+
+        BaseData.__init__(self)
+
+        self.power_spectra = None
+
+
+    @property
+    def has_data(self):
```

Comment:
They do get inherited! In this case, this overrides the previous version to check the `power_spectra` field, instead of the `power_spectrum` field (2D version vs 1D version)

Reply:
Oh whoops, missed that difference!